src/compiler/checker.ts

```diff
@@ -22551,10 +22551,16 @@ namespace ts {
             }
 
             if (entityName.parent!.kind === SyntaxKind.JSDocParameterTag) {
-                const parameter = ts.getParameterFromJSDoc(entityName.parent as JSDocParameterTag);
+                const parameter = getParameterFromJSDoc(entityName.parent as JSDocParameterTag);
                 return parameter && parameter.symbol;
             }
 
+            if (entityName.parent!.kind === SyntaxKind.TypeParameter && entityName.parent!.parent!.kind === SyntaxKind.JSDocTemplateTag) {

```

Comment:
We don't compile with strictNullChecks (yet), and line 22553 is the only place in the whole file where we have a non-null expression. I'd rather we not use it here and remove the one above, as when the time comes and we do enable strictNullChecks, we could unintentionally miss a case because we added these too early.